src/ophyd_epics_devices/zebra.py

```diff
@@ -0,0 +1,306 @@
+        self.source_4 = zebra_rw(int, pv_prefix + "_INP4")
+        self.invert = zebra_rw(int, pv_prefix + "_INV")
+        super().__init__(prefix, name)
+
+    @property
+    def sources(self):
+        return [self.source_1, self.source_2, self.source_3, self.source_4]
+
+
+def boolean_array_to_integer(values: List[bool]) -> int:
+    """Converts a boolean array to integer by interpretting it in binary with LSB 0 bit
+    numbering.
+    Args:
+        values (List[bool]): The list of booleans to convert.
+    Returns:
+        int: The interpretted integer.
+    """
+    return sum(v << i for i, v in enumerate(values))
+
+
+class GateType(Enum):
+    AND = "AND"
+    OR = "OR"
+
+
+class LogicGateConfiguration:
+    NUMBER_OF_INPUTS = 4
+
+    def __init__(self, input_source: int, invert: bool = False) -> None:
+        self.sources: List[int] = []
+        self.invert: List[bool] = []
+        self.add_input(input_source, invert)
+
+    def add_input(
+        self, input_source: int, invert: bool = False
+    ) -> "LogicGateConfiguration":
+        """Add an input to the gate. This will throw an assertion error if more than 4
+        inputs are added to the Zebra.
+        Args:
+            input_source (int): The source for the input (must be between 0 and 63).
+            invert (bool, optional): Whether the input should be inverted. Default
+                False.
+        Returns:
+            LogicGateConfiguration: A description of the gate configuration.
+        """
+        assert len(self.sources) < 4
+        assert 0 <= input_source <= 63
+        self.sources.append(input_source)
+        self.invert.append(invert)
+        return self
+
+    def __str__(self) -> str:
+        input_strings = []
+        for input, (source, invert) in enumerate(zip(self.sources, self.invert)):
+            input_strings.append(f"INP{input+1}={'!' if invert else ''}{source}")
+
+        return ", ".join(input_strings)
+
+
+class LogicGateConfigurer(StandardReadable):
+    DEFAULT_SOURCE_IF_GATE_NOT_USED = 0
+
+    def __init__(self, prefix: str, name=""):
+        self.and_gate_1 = GateControl(prefix, "AND1", name)
+        self.and_gate_2 = GateControl(prefix, "AND2", name)
+        self.and_gate_3 = GateControl(prefix, "AND3", name)
+        self.and_gate_4 = GateControl(prefix, "AND4", name)
+
+        self.or_gate_1 = GateControl(prefix, "OR1", name)
+        self.or_gate_2 = GateControl(prefix, "OR2", name)
+        self.or_gate_3 = GateControl(prefix, "OR3", name)
+        self.or_gate_4 = GateControl(prefix, "OR4", name)
+
+        self.all_gates = {
+            GateType.AND: [
+                self.and_gate_1,
+                self.and_gate_2,
+                self.and_gate_3,
+                self.and_gate_4,
+            ],
+            GateType.OR: [
+                self.or_gate_1,
+                self.or_gate_2,
+                self.or_gate_3,
+                self.or_gate_4,
+            ],
+        }
```

Comment:
Should: Same as above, we can do this all in one and not have to have the individual signals, then probably rename `all_gates` to just `gates`

Reply:
I've now chnaged this, so that we have `zebra.and_gates` and `zebra.or_gates`. I'm resolving this as we no longer have individual signals like above, but you may still want to look at how it's currently implemented which would be:

`zebra.or_gates[1]`

for example.